pkg/network/dns_snooper.go

```diff
@@ -184,11 +168,17 @@ func (s *SocketFilterSnooper) processPacket(data []byte, ts time.Time) {
 	} else {
 		atomic.AddInt64(&s.queries, 1)
 	}
+
+	return nil
 }
 
 func (s *SocketFilterSnooper) pollPackets() {
 	for {
-		data, captureInfo, err := s.source.ZeroCopyReadPacketData()
+		err := s.source.VisitPackets(s.processPacket)
+
+		if err != nil {
+			log.Warnf("error reading packet: %s", err)
+		}
 
 		// Properly synchronizes termination process
 		select {
```

Comment:
So this will not break the loop in `VisitPackets`? You could have `processPacket` return an error if `s.exit` has closed.

Reply:
Good idea, done

Reply:
We could also do
```go
err := s.source.VisitPackets(func(data []byte, timestamp time.Time) error {
  select {
    case <-s.exit:
      return fmt.Errorf("exited")
    default:
  }

  return s.ProcessPacket(data, timestamp)
})

Reply:
We could, but the `exit` logic will translate nicely to windows, so I prefer to leave it as is.